src/query/sql/src/executor/physical_plan_builder.rs

```diff
@@ -1132,9 +1114,8 @@ impl PhysicalPlanBuilder {
 
                     let expr = cast_expr_to_non_null_boolean(expr)?;
 
-                    let (expr, _) = ConstantFolder::fold(&expr, &self.func_ctx, &BUILTIN_FUNCTIONS);
```

Comment:
We can't remove these codes because the `rule_fold_constants` did not fold the whole expr (it only fold the sub expr).